app/helpers/orchestration/orchestrator.rb

```diff
@@ -0,0 +1,168 @@
+    base_configuration = workflow.nil? ? PLAYBOOKS.dig(playbook, :general, :base, option) : PLAYBOOKS.dig(playbook, :workflows, workflow, :base, option)
+    custom_options = workflow.nil? ? PLAYBOOKS.dig(playbook, :general, :custom_options,
+                                                   option) : PLAYBOOKS.dig(playbook, :workflows, workflow, :custom_options, option)
+
+    return if base_configuration.nil?
+    return base_configuration if custom_options.nil?
+
+    nested_configuration(base_configuration, custom_options)
+  end
+
+  # 'base': Return the whole base configuration. Do not follow nested configurations
+  # 'all': Return the base configuration options, but also follow the nested configurations
+  # 'subset': Select the given keys from the options, but also follow the nested configurations
+  # 'remove': Reject the given keys from the options, but also follow the nested configurations
+  # 'custom': Overwrite the whole configuration
+  def nested_configuration(base_configuration, playbook_configuration)
+    # Replace label first if available
+    unless base_configuration[:label].nil?
+      base_configuration[:label] =
+        playbook_configuration[:label].nil? ? base_configuration[:label] : playbook_configuration[:label]
+    end
+    selected_configuration = {}
+    selected_configuration[:label] = base_configuration[:label] if base_configuration[:label].present?
+
+    # If playbook has no type, then we'll assume everything since we don't know what to do
+    type = playbook_configuration[:type].nil? ? 'base' : playbook_configuration[:type]
+
+    case type
+    when 'base'
+      # Return entirety of the base configuration
+      return base_configuration
+    when 'all'
+      # Select the base configuration
+      selected_configuration = base_configuration
+    when 'subset'
+      # If a subset, then look into the configuration and only return the wanted fields
+      selected_configuration[:options] = if base_configuration[:options].is_a?(Array)
+                                           base_configuration[:options].select { |key| playbook_configuration[:config][:set].include?(key) }
+                                         else
+                                           base_configuration[:options].select { |key, _value| playbook_configuration[:config][:set].include?(key) }
+                                         end
+    when 'remove'
+      # If remove, then look into the configuration and reject the listed fields
+      selected_configuration[:options] = if base_configuration[:options].is_a?(Array)
+                                           base_configuration[:options].reject { |key| playbook_configuration[:config][:set].include?(key) }
+                                         else
+                                           base_configuration[:options].reject { |key, _value| playbook_configuration[:config][:set].include?(key) }
+                                         end
```

Comment:
```suggestion
      selected_configuration[:options] = base_configuration[:options].reject { |key| playbook_configuration[:config][:set].include?(key) }
```

Reply:
Much cleaner! Fixing that now.

Reply:
https://github.com/SaraAlert/SaraAlert/blob/985cc43322dd9e8743a0b6e5935220a2b1155f90/app/helpers/orchestration/orchestrator.rb#L58